relay-metrics/benches/benches.rs

```diff
@@ -1,135 +1,197 @@
-            num_metrics: 100,
-            num_metric_names: 100,
-            bucket: counter.clone(),
-            num_project_keys: 1,
-        },
-        MetricInput {
-            num_metrics: 1000,
-            num_metric_names: 1000,
-            bucket: counter.clone(),
-            num_project_keys: 1,
-        },
-        // scaling num_project_keys
-        MetricInput {
-            num_metrics: 100,
-            num_metric_names: 1,
-            bucket: counter.clone(),
-            num_project_keys: 100,
-        },
-        MetricInput {
-            num_metrics: 1000,
-            num_metric_names: 1,
-            bucket: counter,
-            num_project_keys: 1000,
-        },
+    let inputs = vec![
+        (
+            "multiple metrics of the same project",
+            BucketsGenerator {
+                percentage_backdated: 0.5,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..1),
+                timestamp_shifts_generator: NumbersGenerator::new(1..10),
+            },
+        ),
+        (
+            "same metric on different projects",
+            BucketsGenerator {
+                percentage_backdated: 0.5,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..1),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(1..10),
+            },
+        ),
+        (
+            "all backdated metrics",
+            BucketsGenerator {
+                percentage_backdated: 1.0,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
+        (
+            "all non-backdated metrics",
+            BucketsGenerator {
+                percentage_backdated: 0.0,
+                num_buckets: 100_000,
+                metric_ids_generator: NumbersGenerator::new(1..100),
+                project_keys_generator: NumbersGenerator::new(1..100),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
+        (
+            "a lot of metrics from a lot of projects with mix of backdated buckets and non",
+            BucketsGenerator {
+                percentage_backdated: 0.5,
+                num_buckets: 1_000_000,
+                metric_ids_generator: NumbersGenerator::new(1..1_000_000),
+                project_keys_generator: NumbersGenerator::new(1..1_000_000),
+                timestamp_shifts_generator: NumbersGenerator::new(10..50),
+            },
+        ),
     ];
 
-    for input in &inputs {
+    for (input_name, input) in &inputs {
         c.bench_with_input(
```

Comment:
Can make a criterion group here and specify throughput, compare to here https://github.com/getsentry/relay/blob/4d5941afc9544700e6a0b04cff861cf56455cb84/relay-cardinality/benches/redis_impl.rs#L198

Reply:
Oh nice!